src/products/src/products-service/handlers.go

```diff
@@ -162,3 +164,65 @@ func ProductFeatured(w http.ResponseWriter, r *http.Request) {
 func enableCors(w *http.ResponseWriter) {
 	(*w).Header().Set("Access-Control-Allow-Origin", "*")
 }
+
+// Update Stock Quantity for one item Handler
+func UpdateProduct(w http.ResponseWriter, r *http.Request) {
+
+	enableCors(&w)
+
+	var product Product
+	body, err := ioutil.ReadAll(io.LimitReader(r.Body, 1048576))
+	if err != nil {
+		panic(err)
+	}
+	if err := r.Body.Close(); err != nil {
+		panic(err)
+	}
+	if err := json.Unmarshal(body, &product); err != nil {
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		w.WriteHeader(422) // unprocessable entity
+		if err := json.NewEncoder(w).Encode(err); err != nil {
+			panic(err)
+		}
+	}
+
+	log.Println("UpdateProducts  ", product)
+
+	ret := RepoUpdateProduct(product)
```

Comment:
How will product add vs update work? If an `id` is specified, should an existing product be updated. If `id` is absent, add a new product?

Reply:
I propose:

Update existing:

		"PUT",
		"/products/{id}",
		UpdateProduct,
---
create new

		"POST",
		"/products/",
		NewProduct,

---
update stock

		"PUT",
		"/products/inventory/{id}",
		UpdateInventory,

Reply:
Looks good to me.

Another thought on adding products, which doesn't necessarily need to be solved in this PR, is how product images will be added. Perhaps this is solved when the UI changes are introduced to add/update products where an image upload to S3 (and resize?) is supported.

Reply:
For images for now you need 2 steps when creating a new product via API:
- upload 1 image to the relevant folder on s3
- PUT json product data with reference to the image filename in the payload

I would see these 2 operations done from the UI perspective. 

For a later PR